spyder/plugins/ipythonconsole.py

```diff
@@ -907,7 +909,7 @@ def write_to_stdin(self, line):
 
     @Slot()
     @Slot(bool)

```

Comment:
This needs to be

    @Slot(str)
    @Slot(bool, str)

and you need to change below

    filename=''

instead of

    filename=None